src/services/blocks/trace/Trace.ts

```diff
@@ -0,0 +1,666 @@
+import { ApiPromise } from '@polkadot/api';
+import { AccountInfo } from '@polkadot/types/interfaces';
+import { Registry } from '@polkadot/types/types';
+import { stringCamelCase } from '@polkadot/util';
+import BN from 'bn.js';
+import { InternalServerError } from 'http-errors';
+
+import { IOption, isSome } from '../../../types/util';
+import {
+	ActionGroup,
+	BlockTrace,
+	KeyInfo,
+	Operation,
+	PAccountEvent,
+	PActionEvent,
+	PalletKeyInfo,
+	PEvent,
+	Phase,
+	PhaseOther,
+	SpanWithChildren,
+	SpecialKeyInfo,
+	TraceSpan,
+} from './types';
+
+const EMPTY_KEY_INFO = {
+	error: 'key not found',
+};
+
+/**
+ * Known spans
+ */
+const SPANS = {
+	executeBlock: {
+		name: 'execute_block',
+		target: 'frame_executive',
+	},
+	applyExtrinsic: {
+		name: 'apply_extrinsic',
+		target: 'frame_executive',
+	},
+	onInitialize: {
+		name: 'on_initialize',
+		// Targets vary by pallet
+	},
+	onFinalize: {
+		name: 'on_finalize',
+		// Targets vary by pallet
+	},
+};
+
+// In the future this will need a more chain specific solution that creates
```

Comment:
Should we make an issue for this?

Reply:
https://github.com/paritytech/substrate-api-sidecar/issues/528